tests/foreman/api/test_subnet.py

```diff
@@ -498,6 +497,31 @@ def test_positive_list_parameters(self):
             parameters
         """
 
+        parameter = {'name': gen_string('alpha'), 'value': gen_string('alpha')}
+        org = entities.Organization().create()
+        loc = entities.Location(organization=[org]).create()
+        org_subnet = entities.Subnet(
+            location=[loc],
+            organization=[org],
+            ipam=u'DHCP',
+            vlanid=gen_string('numeric', 3),
+            subnet_parameters_attributes=[parameter]
+        ).create()
+        self.assertEqual(org_subnet.subnet_parameters_attributes[0]['name'],
+                         parameter['name'])
+        self.assertEqual(org_subnet.subnet_parameters_attributes[0]['value'],
+                         parameter['value'])
+        sub_param = entities.Parameter(
+            name=gen_string('alpha'),
+            subnet=org_subnet.id,
+            value=gen_string('alpha')
+        ).create()
+        org_subnet = entities.Subnet(id=org_subnet.id).read()
+        params_list = [{param['name']: param['value']}
+                       for param in org_subnet.subnet_parameters_attributes
+                       if param['name'] == sub_param.name]
```

Comment:
Why list of dicts?
You can have it just as a dict:
```python
        params_list = {param['name']: param['value']
                       for param in org_subnet.subnet_parameters_attributes
                       if param['name'] == sub_param.name}
```

This will give you structure like:
```
{'random_param_name': 'this_param_value'}
```

And then, assertion below can be just:
```python
self.assertEqual(params_list[sub_param.name], sub_param.value)
```